pkg/operator/csi/csicontrollerset/csi_controller_set.go

```diff
@@ -38,25 +36,12 @@ type CSIControllerSet struct {
 	csiDriverNodeServiceController       factory.Controller
 	serviceMonitorController             factory.Controller
 
-	preRunCachesSynced []cache.InformerSynced
-	operatorClient     v1helpers.OperatorClient
-	eventRecorder      events.Recorder
+	operatorClient v1helpers.OperatorClient
+	eventRecorder  events.Recorder
 }
 
 // Run starts all controllers initialized in the set.
 func (c *CSIControllerSet) Run(ctx context.Context, workers int) {
-	defer utilruntime.HandleCrash()
-
-	// Create a custom context to sync informers added  with .WithExtraInformers().
-	// This context is not used for individual controllers because factory.Factory
-	// will overwrite the timeout value.
-	cacheSyncCtx, cacheSyncCancel := context.WithTimeout(ctx, defaultCacheSyncTimeout)
-	defer cacheSyncCancel()
-
-	if !cache.WaitForCacheSync(cacheSyncCtx.Done(), c.preRunCachesSynced...) {
```

Comment:
interesting.  pushing it down to the controllers is ok, but do they know to do it?